pkg/tarball/tarball.go

```diff
@@ -83,3 +85,76 @@ func DecodeTarball(reader io.Reader, baseDir string) error {
 
 	return nil
 }
+
+// TarDir tars up an entire directory and outputs the tarball to the specified output path.
+// If useGzip is true, it also gzips the resulting tarball.
+func TarDir(dir, outpath string, useGzip bool) error {
```

Comment:
I probably should've done this instead of implementing whatever I wrote in https://github.com/vmware-tanzu/sonobuoy/blob/master/cmd/sonobuoy/app/splat.go 😅

Reply:
nit: I prefer the name `DirToTarball` because `TarDir` feels like it can go either way of tar _to_ dir or tar _from_ dir (or is it second-language thing?)

Reply:
/shrug, its fine either way but the 'DirToTarball' is even more explicit. I can change that.